File: server/feed5/swagger_server/daos/outbound_dao.py
from ..models.outbound import Outbound
from .cassandra_dao import CassandraDAO

class Outbound(CassandraDAO):
    _from = None
    subject = None
    story = None
    insert = 'insert_outbound'
    query = 'select_outbound'

    def __init__(self, _from: int = None, subject: str = None, story: str = None):
        self._from = _from
        self.subject = subject
        self.story = story
        self.prepare(self.insert, 'insert into Outbound (ParticipantID, Occurred, Subject, Story) values (?, now(), ?, ?)')
        self.prepare(self.query, 'select toTimestamp(occurred) as occurred, subject, story from Outbound where participantid = ? order by occurred desc')

    def save(self):
        self.execute(self.insert, [ self._from, self.subject, self.story ])

    def makeOutbound(self, dikt):
        retVal = {}
        retVal['occurred'] = dikt['occurred'].strftime('%Y-%m-%d')
        retVal['from'] = self._from
        retVal['subject'] = dikt['subject']
        retVal['story'] = dikt['story']
        return retVal

    def load(self):
        return self.execute(self.query, [ self._from ], self.makeOutbound)

    def __repr__(self):
        return '{"from":%d, "subject":"%s", "story":"%s"}' % (self._from, self.subject, self.story)

    def _from(self) -> int:
        return self._from

    def subject(self) -> str:
        return self.subject

    def story(self) -> str:
        return self.story

File: server/feed11/src/main/scala/info/glennengstrand/news/dao/OutboundDao.scala
package info.glennengstrand.news.dao

import scala.concurrent.Future
import scala.concurrent._
import ExecutionContext.Implicits.global
import scala.collection.JavaConverters._

import com.datastax.oss.driver.api.core.cql.{BoundStatement, PreparedStatement}
import info.glennengstrand.news.model.Outbound

class OutboundDao extends DataAccess[Outbound] {
  private val selectCql = "select toTimestamp(occurred) as occurred, subject, story from Outbound where participantid = ? order by occurred desc"
  private val insertCql = "insert into Outbound (ParticipantID, Occurred, Subject, Story) values (?, now(), ?, ?)"
  private lazy val insertStmt = CassandraDao.session.prepare(insertCql)
  private lazy val selectStmt = CassandraDao.session.prepare(selectCql)
  
  override def fetchSingle(id: Int): Future[Outbound] = {
    Future(Outbound(None, None, None, None))
  }
  override def insert(ob: Outbound): Future[Outbound] = {
    val bs = insertStmt.bind(new java.lang.Integer(extractId(ob.from.get.toString.asInstanceOf[String]).toInt), ob.subject.get.toString.asInstanceOf[String], ob.story.get.toString.asInstanceOf[String])
    CassandraDao.session.execute(bs)
    Future(ob)
  }
  override def fetchMulti(id: Int): Future[Seq[Outbound]] = {
    val bs = selectStmt.bind(id.asInstanceOf[Object])
    val retVal = for {
      r <- CassandraDao.session.execute(bs).iterator().asScala
    } yield Outbound(Option(toLink(id.toLong)), Option(r.getInstant(0).toString()), Option(r.getString(1)), Option(r.getString(2)))
    Future(retVal.toSeq)
  }
}